scripts/lib/CIME/build.py

```diff
@@ -457,10 +465,15 @@ def _case_build_impl(caseroot, case, sharedlib_only, model_only, buildlist,
     os.environ["CISM_USE_TRILINOS"]    = stringify_bool(cism_use_trilinos)
     os.environ["MALI_USE_ALBANY"]      = stringify_bool(mali_use_albany)
     os.environ["USE_MOAB"]             = stringify_bool(use_moab)
+    os.environ["PIO_VERSION"]          = str(case.get_value("PIO_VERSION"))
```

Comment:
Making sure everything the Makefile might need is set in the environment so it doesn't have to use xmlquery.

Reply:
It doesn't in general use xmlquery and it doesn't need to use environment variables - most builds use command line arguments to make eg:
gmake PIO_VERSION=2

Reply:
I made a big effort to not rely on the environment for Makefile arguments - I really don't want to see that reversed.